nomad/keyring_endpoint.go

```diff
@@ -138,20 +142,29 @@ func (k *Keyring) Update(args *structs.KeyringUpdateRootKeyRequest, reply *struc
 		return err
 	}
 
+	// make sure it's been added to the local keystore before we write
+	// it to raft, so that followers don't try to Get a key that
+	// hasn't yet been written to disk
+	err = k.encrypter.AddKey(args.RootKey)
+	if err != nil {
+		return err
+	}
+
 	// unwrap the request to turn it into a meta update only
```

Comment:
The RPC updates both, by calling `AddKey` with the key material and then `raftApply` with just the metadata. But maybe I'm misunderstanding how you're suggesting to update the comment.

Reply:
I missed they write to the keyring, so I only saw the transition to meta-only. 🤦  In that case, the current state of `Update` seems dangerous since it can alter a key that has been used for encryptions. However, I imagine the defense against that is coming when we add the key usage accounting?

Reply:
Done in 3ddfc84c0fda0830ff9b579c5eee2cfe62251d55